node/network/collator-protocol/src/collator_side.rs

```diff
@@ -240,25 +241,86 @@ where
 /// Issue a connection request to a set of validators.
 async fn connect_to_validators<Context>(
 	ctx: &mut Context,
+	relay_parent: Hash,
 	state: &mut State,
 	validators: Vec<ValidatorId>,
 ) -> Result<()>
 where
 	Context: SubsystemContext<Message = CollatorProtocolMessage>
 {
+	let (result, revoke) = connect_to_validators_impl(ctx, relay_parent, validators).await?;
+
+	for (validator_id, peer_id) in result.into_iter() {
+		state.known_validators.insert(peer_id, validator_id);
+	}
+
+	// TODO: do we want to revoke requests at all?
+	drop(revoke);
```

Comment:
Again, I think that there sould just be an api to disconnect from nodes we are no longer interested in, not the whole group we connected to.

This would be useful not only here but in the case validators considering to drop evil collators.

Reply:
That would complicate the caller's side, because currently, the networking request operates on `AuthorityDiscoveryId`, not `ValidatorId`. The mapping is abstracted away in the `connect_to_validators_impl` function, which is ideally to be extracted somewhere to be reused by other subsystems. 

If we introduce another Message `DisconnectFromValidators`, this mapping would have to be maintained by every subsystem.

> This would be useful not only here but in the case validators considering to drop evil collators.

we could just revoke the previous connection request and not request it anymore. We also have `ReportPeer` message for this.